crates/rpc/rpc/src/otterscan.rs

```diff
@@ -305,11 +307,141 @@ where
     /// Handler for `ots_searchTransactionsAfter`
     async fn search_transactions_after(
         &self,
-        _address: Address,
-        _block_number: u64,
-        _page_size: usize,
-    ) -> RpcResult<TransactionsWithReceipts> {
-        Err(internal_rpc_err("unimplemented"))
+        address: Address,
+        block_number: u64,
+        page_size: usize,
+    ) -> RpcResult<TransactionsWithReceipts<RpcTransaction<Eth::NetworkTypes>>> {
+        let state = self.eth.latest_state().map_err(|e| internal_rpc_err(e.to_string()))?;
+        let account = state.basic_account(&address).map_err(|e| internal_rpc_err(e.to_string()))?;
+
+        if account.is_none() {
+            return Err(EthApiError::InvalidParams(
+                "invalid parameter: address does not exist".to_string(),
+            )
+            .into());
+        }
+
+        let tip: u64 = self.eth.block_number()?.saturating_to();
+
+        if block_number > tip {
+            return Err(EthApiError::InvalidParams(
+                "invalid parameter: block number is larger than the chain tip".to_string(),
+            )
+            .into());
+        }
+
+        let mut txs_with_receipts = TransactionsWithReceipts {
+            txs: Vec::default(),
+            receipts: Vec::default(),
+            first_page: false,
+            last_page: block_number == 0,
+        };
+
+        let filter = TraceFilter {
+            from_block: None,
+            to_block: None,
```

Comment:
I think this is where we'd need to configure the block_number and perhaps the page_size?

because worst case this would trace the entire chain

Reply:
So do you think we should put a limit to the number of blocks traced, or to the number of transactions? If we limit the number of transaction, there's still the chance to trace the entire chain e.g. if the account has 5 transactions in total and the user requests 10.

Reply:
unfortunately, we don't have another way to determine this so this is will always be possible, so we should doing some chunking instead so we limit how many blocks we trace at once, so something like a https://docs.rs/futures/latest/futures/stream/struct.FuturesUnordered.html with limited capacity where we push new block tasks

Reply:
What you mean is processing blocks as batches of something like 100 or 1000 right? E.g. processing 1000 block, waiting until they're all complete, and continue with another 1000 blocks if we didn't reach the page size yet. In that case using `try_join_all` looks like a better idea since we have to wait for all 1000 blocks to complete anyways. It's also how it's done in `trace_filter`.

Reply:
On second thought, FuturesUnordered will be useful

Reply:
Okay I changed the implementation with `FuturesUnordered` and it works much faster now. I used 1000 block batches for now but we can change that. I think we can also put a limit to `page_size` to prevent requesting unreasonable number of transactions. What do you think would be a good number for that?